### ISSUE TITLE
[MDCRipple] The ripple is misaligned when used in dialog

### ISSUE THREAD

**Issue by DRiFTy17** [material-components/material-components-web#5443]
## Bug report
When placing an element that has a ripple attached (icon-button, checkbox, radio... etc.) inside of an MDC dialog, the ripple is misaligned when first opening the dialog. Upon interacting with the element (clicking for example) the ripple repositions itself properly.

After some research it looks to be a timing issue related to the `scale` `transform` style and `transition`. Along with the timing of when the `layoutInternal_` function is called within the ripple foundation. The metrics that is uses are based on the scaled down dialog.

### Steps to reproduce
View the stackblitz here:
https://stackblitz.com/edit/mdc-dialog-ripple-alignment

1. When the dialog opens, hover over the heart icon-button
2. The ripple will not be the correct size, and is misaligned
3. Click the icon-button and it will then reposition properly

### Actual behavior
The ripple position and size is incorrect due to CSS scale transform and transition. 

### Expected behavior
The ripple should be positioned and sized properly in a dialog (or any animating container element) as if it was placed directly on the static page.

### Screenshots
![image](https://user-images.githubusercontent.com/2653457/72086666-4a0cd800-32d5-11ea-918a-68e4e04a65cd.png)

### Your Environment:

| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          | 4.0.0
| Browser          | Chrome
| Operating System | Windows 10

### Additional context
Just wanted to get some thoughts on a possible solution or workaround from you. It is causing some negative feedback from our users and other dev team members.

This does happen with other elements like the checkbox, radio, switch, ... etc. This is a tough one to "fix" because the ripple would either have to know its context element, or delay its positioning logic but that seems brittle.

### Possible solution
I've tried messing around with the `layoutInternal_` function in the ripple foundation by delaying it with `requestAnimationFrame` and everything I've tried seems hacky. I also don't think the dialog or ripple should know about each other, but I have gotten it to work by running the `layout` method on each ripple from the dialog.

Also, just a side note, removing the scale transition in the dialog does cause it to render properly.

**Comment by asyncLiz** [comment]
Thanks for the issue! It looks like we may already be aware of this bug internally, I'll ask around to see if we have an idea of how to fix it.

**Comment by asyncLiz** [comment]
I did a little more digging, and this is actually working as intended. See the [JavaScript instantiation](https://github.com/material-components/material-components-web/tree/master/packages/mdc-dialog#javascript-instantiation) section of the dialog's readme.

TL;DR, instantiate or call `.layout()` on dialog elements when it is opened. Here's a fork with the solution: https://stackblitz.com/edit/mdc-dialog-ripple-alignment-eoogdd?file=index.js

**Comment by DRiFTy17** [comment]
@asyncLiz Ah okay, I had a feeling that would be the case (sorry that I missed that note in the docs). It's slightly cumbersome, but it's also the only way (and non-hack) I have been able to consistently get it working as-is. Thanks for the info, I appreciate it!

**Comment by simonziegler** [comment]
Hi, I'm building a Blazor component library for Material Theme. Is there a way that I can find all relevant elements within a dialog to call layout() for each? Thank you.

**Comment by DouglasRiddle** [comment]
@simonziegler  I address this issue  with the below. Since I don't know if the dialog will contain a troublesome alignment, I check for the existence of the element and then run .layout if it exists. I'd also point out that I'm using 5.1.0 and it gave me a ".layout is not a function" error when using .layout(), but .layout seems to  work fine.

```javascript
dialog.listen( 'MDCDialog:opened', () => {
    const radioElem = document.querySelector( '.mdc-dialog .mdc-radio' );
    const checkboxElem = document.querySelector( '.mdc-dialog .mdc-checkbox' );
    const iconButtonElem = document.querySelector( '.mdc-dialog .mdc-icon-button' );
    const switchElem = document.querySelector( '.mdc-dialog .mdc-switch' );
    if ( radioElem !== null ) { mdc.radio.MDCRadio.attachTo( radioElem ).layout; }
    if ( checkboxElem !== null ) { mdc.checkbox.MDCCheckbox.attachTo( checkboxElem ).layout; }
    if ( iconButtonElem !== null ) { const iconButton = mdc.ripple.MDCRipple.attachTo( iconButtonElem ); iconButton.unbounded = true; iconButton.layout; }
    if ( switchElem !== null ) { mdc.switchControl.MDCSwitch.attachTo( switchElem ).layout; }
} );
```
